### ISSUE TITLE
Reader comment failed

### ISSUE THREAD

**Issue by phamdacloc** [wordpress-mobile/WordPress-iOS#4369]
There's an error that occur when reply to a nested comment in Reader because the commentID is nil.

Steps to recreate:
1. Open a Reader's comment view controller.
2. Comment the post.
3. Without returning to the previous screen, reply to the comment made in step 2.

The culprit lies in the following function:

```php
- (void)replyToPostWithID:(NSNumber *)postID
                   siteID:(NSNumber *)siteID
                  content:(NSString *)content
                  success:(void (^)())success
                  failure:(void (^)(NSError *error))failure
{
    // Create and optimistically save a comment, based on the current wpcom acct
    // post and content provided.
    Comment *comment = [self createHierarchicalCommentWithContent:content withParent:nil postID:postID siteID:siteID];
    NSManagedObjectID *commentID = comment.objectID;
    void (^successBlock)(RemoteComment *remoteComment) = ^void(RemoteComment *remoteComment) {
        [self.managedObjectContext performBlock:^{
            Comment *comment = (Comment *)[self.managedObjectContext existingObjectWithID:commentID error:nil];
            if (!comment) {
                return;  // <--- Since it returned here, the new comment ID was never updated in DB.
            }
            // Update and save the comment
            [self updateCommentAndSave:comment withRemoteComment:remoteComment];
            if (success) {
                success();
            }
        }];
    };
```

**Comment by aerych** [comment]
Thanks for the report @phamdacloc!  
Patch coming.

**Comment by kurzee** [comment]
Merged into 5.7
